packages/hooks/src/data/QueryData.ts

```diff
@@ -118,6 +115,85 @@ export class QueryData<TData, TVariables> extends OperationData {
     delete this.previousData.result;
   }
 
+  public getOptions() {
+    const options = super.getOptions();
+    const lazyOptions = this.lazyOptions || {};
+    const updatedOptions = {
+      ...options,
+      variables: {
+        ...options.variables,
+        ...lazyOptions.variables
+      },
+      context: {
+        ...options.context,
+        ...lazyOptions.context
+      }
+    };
+
+    // Triggering a query in lazy mode overrides `skip` settings.
+    if (options.lazy) {
+      updatedOptions.skip = false;
+    }
+
+    return updatedOptions;
+  }
+
+  private initialLazyResult(): QueryTuple<TData, TVariables> {
+    return [
+      {
+        loading: false,
+        networkStatus: NetworkStatus.ready,
+        called: false,
+        data: undefined
+      } as QueryResult<TData, TVariables>,
+      this.executeLazy.bind(this)
+    ];
+  }
+
+  private executeLazy(options?: QueryLazyOptions<TVariables>) {
+    this.runLazy = true;
+    this.lazyOptions = options;
+    this.forceUpdate();
+  }
+
+  private getExecuteResult(): TLazy extends undefined
+    ? QueryResult<TData, TVariables>
+    : QueryTuple<TData, TVariables> {
+    const result = this.getQueryResult();
+    this.startQuerySubscription();
+    return (this.getOptions().lazy !== undefined
+      ? [result, this.executeLazy.bind(this)]
+      : result) as any;
+  }
+
+  private getExecuteSsrResult(): TLazy extends undefined
```

Comment:
If we don’t keep the `TLazy` approach, I think `getExecuteResult` and `getExecuteSsrResult` should stick to returning just the result, so we don’t have to juggle different return types.